src/utils/require-template.js

```diff
@@ -19,30 +19,34 @@ function isValidTemplate(template){
   ].every((method) => template[method]);
 }
 
+const getOcTemplate = (path) => {
+  if (require.cache && !!require.cache[path]) {
+    delete require.cache[path];
+  }
+  return require(path);
+};
 
-module.exports = function(template) {
+module.exports = function (template) {
   let ocTemplate;
   const localTemplate = path.join(__dirname, '../../', 'node_modules', template);
   const relativeTemplate = path.resolve('.', 'node_modules', template);
 
   try {
-    if (require.cache && !!require.cache[localTemplate]) {
-      delete require.cache[localTemplate];
-    }
-    ocTemplate = require(localTemplate);
-  } catch(err) {
+    ocTemplate = getOcTemplate(template);
```

Comment:
Hi,
Yes I moved the logic into a function but I've also added an additional check of getting the template by name (line 35). 
Try to do this: install OC globally w/ yarn and make sure your publish doesn't work. Then replace require-template.js in the yarn OC global folder w/ the file I've changed and see if that fixes. Thanks!